SecretManagement.LastPass.Extension/SecretManagement.LastPass.Extension.psm1

```diff
@@ -70,35 +107,51 @@ function Invoke-lpass {
     
     $UseWSL = $AdditionalParameters.wsl -eq $true
     $lpassPath = if ($null -ne $AdditionalParameters.lpassPath) { "`"$($AdditionalParameters.lpassPath)`"" } else { 'lpass' }
-
-    $IgnoreErrors = $ErrorActionPreference -eq 'SilentlyContinue'
     $UseNative = -not $UseWSL
 
     if (($UseNative -and -not (Get-Command $lpassPath -EA 0)) -or 
         ($UseWSL -and (& wsl $lpassPath --version ) -notlike 'LastPass CLI*')) {
         throw "lpass executable not found or installed."
     }
 
-    # All other implementations seemed to succeed on 1 or more platform but failed when considering Windows (Wsl) / Linux / Mac together
-    if ($InputObject) {
-        switch ($true) {
-            {$UseNative -and $IgnoreErrors}  {$result = $InputObject | & $lpassPath @Arguments 2>&1; break}
-            {$UseNative}                     {$result = $InputObject | & $lpassPath @Arguments; break}
-            {$UseWSL -and $IgnoreErrors}     {$result = $InputObject | & wsl $lpassPath @Arguments 2>&1; break}
-            {$UseWSL}                        {$result = $InputObject | & wsl $lpassPath @Arguments; break}
-        }
-    } else {
-        switch ($true) {
-            { $UseNative -and $IgnoreErrors }   {$result = & $lpassPath @Arguments 2>&1; break}
-            { $UseNative }                      {$result = & $lpassPath @Arguments; break}
-            { $UseWSL -and $IgnoreErrors }      {$result = & wsl $lpassPath @Arguments 2>&1; break}
-            { $UseWSL }                         {$result = & wsl $lpassPath @Arguments; break}
+    $Params = @{
+        InputObject = $InputObject
+        UseNative   = $UseNative
+        lpassPath   = $lpassPath
+        Arguments   = $Arguments
+    }
+
+    # If we do redirect on the command themselves, it doesn't work.
+    # Doing redirect on the commands wrapped in a function work.
+    
+    if ($Arguments.Count -gt 0) {
+        switch ($Arguments[0]) {
+            'login' {  
+                # We want the prompt always, so no redirect
+                $result = Invoke-lpassInternal @Params
+            }
+            {$_ -in 'show','ls','rm'} {
+                # We might want the prompt, but are not sure yet.
+                $result = Invoke-lpassInternal @Params 2>&1
+                # If we get the message stating we might be logged out, we reissue the command without redirect (for Prompt)
+                if ($result -is [System.Management.Automation.ErrorRecord] -and [String]$result -like $lpassMessage.LoggedOut) {
+                    $result2 = Invoke-lpassInternal @Params
+                    # If $result2 -eq $null, something will have been printed in the console (because we disabled the redirect)
```

Comment:
Can we still throw that SecretManagement special exception when they didn't put their password in so that a user could theoretically use a try/catch around Set-Secret?

Reply:
@TylerLeonhardt If I get your gist, this is what line 142 address. 

```
 if ($null -ne $result2 -or $Arguments[0] -eq 'rm') { $result = $result2 }
```

Basically, we do this : 

1. First call (redirect) is made
2. Check for logged out is made
3. Call is made again without suppression of CLI native error (to allow prompt)
4. Line #142

If the command is rm, success is assumed (This is because rm return $null)
If the command is anything else (show, ls) , then a $null check is made.
- If $result2 is $null, we do nothing so $result (which contains the original logged out redirected error) is the one evaluated later and an error is thrown to this effect.

- If $result2 is not $null, we assign its value to $result so an error is NOT thrown later and everything continue.

As for `Set-Secret` There's always a `show` first to determine if we are in add or edit mode.
Thus, the prompt is evaluated then and since we throw an error, the Set secret never go past the show operation and work properly.

Reply:
Ahhh I see! Ok :)